### PR TITLE
(Feature) Verify user facebook email

### PR DESCRIPTION
**Pull request by serdiukov-o-nordwhale**
# Description

- refactor proof verification and torusVerifier
- add facebookVerifier
- add email verification by accessToken

About #185 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/server/storage/storageAPI.js [hunk 1]
```diff
@@ -28,29 +27,37 @@ const setup = (app: Router, gunPublic: StorageAPI, storage: StorageAPI) => {
...
+
+      // if torus, then we first verify the user mobile/email by verifying it matches the torus public key
+      // (torus maps identifier such as email and mobile to private/public key pairs)
+      if (torusProof || torusAccessToken) {
+        const verifier = createUserVerifier(userRecord, userPayload, logger)
 
-        logger.info('TorusVerifier result:', { emailVerified, mobileVerified })
-        userRecord.smsValidated |= mobileVerified
-        userRecord.isEmailConfirmed |= emailVerified
+        if (torusProvider == 'facebook') {
```
Comment (sirpy) [inline src/server/storage/storageAPI.js:51]: it was better when it was encapsulated inside the verifier

File: src/server/storage/storageAPI.js [hunk 2]
```diff
@@ -28,29 +27,37 @@ const setup = (app: Router, gunPublic: StorageAPI, storage: StorageAPI) => {
...
-          logger.warn('TorusVerifier failed:', { e, msg: e.message })
-          return { emailVerified: false, mobileVerified: false }
-        })
+      logger.debug('new user request:', { data: userPayload, userRecord })
+
+      const {
+        torusProof,
+        torusProvider,
+        torusProofNonce,
+        torusAccessToken,
```
Comment (yaroslav-fedyshyn-nordwhale) [inline src/server/storage/storageAPI.js:39]: are you sure the accessToken accessible within 'torus' prefix?

Comment (serdiukov-o-nordwhale) [reply src/server/storage/storageAPI.js:39]: https://github.com/GoodDollar/GoodDAPP/pull/2070/files#diff-1e67442d754aab36afd1a9966e06c497R376

File: src/server/storage/storageAPI.js (hunk 1 above)
Comment (serdiukov-o-nordwhale) [reply src/server/storage/storageAPI.js:51]: generally we have now 2 totally different ways:
- via proof
- via access token

they have different input params and different result
so they couldn't be processed using the single service class (e.g. TorusVerifier)

I've added ./verifier that incapsulates both flows.
We could move torusProvider == 'facebook' check inside

**Pull request merged**